File: src/Web.Admin/Models/AuthModels.cs
using System.ComponentModel.DataAnnotations;

namespace BarbeariaSaaS.Web.Admin.Models;

public class User
{
    public string Id { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public UserRole Role { get; set; }
    public string TenantId { get; set; } = string.Empty;
    public bool IsActive { get; set; }
    public DateTime CreatedAt { get; set; }
}

public enum UserRole
{
    Admin,
    Barbeiro,
    Recepcionista,
    Cliente
}

public class LoginRequest
{
    [Required(ErrorMessage = "Email é obrigatório")]
    [EmailAddress(ErrorMessage = "Email inválido")]
    public string Email { get; set; } = string.Empty;

    [Required(ErrorMessage = "Senha é obrigatória")]
    [MinLength(6, ErrorMessage = "Senha deve ter pelo menos 6 caracteres")]
    public string Password { get; set; } = string.Empty;
}

public class AuthResult
{
    public bool Success { get; set; }
    public User? User { get; set; }
    public string? Token { get; set; }
    public string? RefreshToken { get; set; }
    public AuthError? Error { get; set; }
    public string? Message { get; set; }
}

public class AuthState
{
    public bool IsAuthenticated { get; set; }
    public User? User { get; set; }
    public string? Token { get; set; }
    public string? RefreshToken { get; set; }
    public DateTime? TokenExpiry { get; set; }
}

public enum AuthError
{
    InvalidCredentials,
    UserNotFound,
    UserInactive,
    TokenExpired,
    NetworkError,
    UnknownError
}

File: src/Web.Mobile/src/app/core/models/auth.models.ts
export interface User {
  id: string;
  email: string;
  name: string;
  role: UserRole;
  tenantId: string;
  isActive: boolean;
  createdAt: Date;
  phone?: string;
}

export enum UserRole {
  ADMIN = 'admin',
  BARBEIRO = 'barbeiro',
  RECEPCIONISTA = 'recepcionista',
  CLIENTE = 'cliente'
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface AuthResult {
  success: boolean;
  user?: User;
  token?: string;
  refreshToken?: string;
  error?: AuthError;
  message?: string;
}

export interface AuthState {
  isAuthenticated: boolean;
  user: User | null;
  token: string | null;
  refreshToken: string | null;
  tokenExpiry: Date | null;
}

export enum AuthError {
  INVALID_CREDENTIALS = 'invalid_credentials',
  USER_NOT_FOUND = 'user_not_found',
  USER_INACTIVE = 'user_inactive',
  TOKEN_EXPIRED = 'token_expired',
  NETWORK_ERROR = 'network_error',
  UNKNOWN_ERROR = 'unknown_error'
}

export interface RegisterClienteRequest {
  nome: string;
  email: string;
  senha: string;
  telefone: string;
  tenantId?: string; // Will be selected from available barbershops
}